modules/tinymce/src/core/test/ts/browser/UndoManagerTest.ts

```diff
@@ -611,4 +612,82 @@ describe('browser.tinymce.core.UndoManagerTest', () => {
 
     assert.equal(changeEventCounter, 0, 'No events should be detected');
   });
+
+  context('dispatchChange', () => {
+    const initialContent = '<p>some inital content</p>';
+    const manualModifiedLevelContent = 'a modified last level';
+
+    const assertChangeEvent = (
+      eventContent: { levelContent: string; lastLevelContent: string | undefined },
+      expectedLevelContent: string | undefined,
+      expectedLastlevelContent: string | undefined
+    ) => {
+      assert.deepEqual(eventContent, {
+        levelContent: expectedLevelContent,
+        lastLevelContent: expectedLastlevelContent
+      }, 'the level or the last level have not the expected content');
+    };
+
+    let changeEventCounter: number;
+    let currentChangeEvent: { levelContent: string; lastLevelContent: string } | undefined;
+
+    const onChange = (e: EditorEvent<{
+      level: UndoLevel;
+      lastLevel: UndoLevel | undefined;
+    }>) => {
+      changeEventCounter++;
+      currentChangeEvent = { levelContent: e.level?.content, lastLevelContent: e.lastLevel?.content };
+    };
+
+    beforeEach(() => {
+      changeEventCounter = 0;
+      currentChangeEvent = undefined;
+    });
+
+    it('TINY-8641: Dispatch change with current editor status as level and current undoManager layer as lastLevel', () => {
+      const editor = hook.editor();
+
+      editor.resetContent(initialContent);
+
+      editor.on('change', onChange);
```

Comment:
I guess this binding/unbinding could in theory be a beforeEach/afterEach thing since the onChange is a function outside these tests.